hail/src/main/scala/is/hail/backend/spark/LowerTableIR.scala

```diff
@@ -114,14 +114,14 @@ object LowerTableIR {
       val row = Ref(genUID(), child.typ.rowType)
       val global = loweredChild.globals
       val env: Env[IR] = Env("row" -> row, "global" -> Ref(global.name, global.value.typ))
-      loweredChild.copy(body = ArrayFilter(loweredChild.body, row.name, Subst(cond, env)))
+      loweredChild.copy(body = ArrayFilter(loweredChild.body, row.name, Subst(cond, BindingEnv(env))))
 
     case TableMapRows(child, newRow) =>
       val loweredChild = lower(child)
       val row = Ref(genUID(), child.typ.rowType)
       val global = loweredChild.globals
       val env: Env[IR] = Env("row" -> row, "global" -> Ref(global.name, global.value.typ))
-      loweredChild.copy(body = ArrayMap(loweredChild.body, row.name, Subst(newRow, env)))
+      loweredChild.copy(body = ArrayMap(loweredChild.body, row.name, Subst(newRow, BindingEnv(env, scan = Some(env)))))
```

Comment:
This actually errors out with scans, since that relies on the (nonexistent) aggregator infrastructure. In either case, at this level, we shouldn't be tracking separate envs for scans and aggregations---they'll have been lowered into something else.